### PR TITLE
Refactor storage tests

### PR DESCRIPTION
**Pull request by lzchen**

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py [hunk 1]
```diff
@@ -81,49 +87,61 @@ def test_constructor(self):
...
+        blob_mock.lease.return_value = True
+        envelope_mock = {"name":"test","time":"time"}
+        blob_mock.get.return_value = [envelope_mock]
+        exporter.storage.gets.return_value = [blob_mock]
+        with mock.patch.object(AzureMonitorClient, 'track') as post:
+            post.return_value = TrackResponse(
+                items_received=1,
+                items_accepted=1,
+                errors=[],
+            )
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py:106]: Could you explain the function or `post`?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py [hunk 2]
```diff
@@ -81,49 +87,61 @@ def test_constructor(self):
         self.assertEqual(base._timeout, 10)
         self.assertEqual(base._api_version, "2021-02-10_Preview")
         self.assertEqual(base._storage_min_retry_interval, 100)
+        self.assertEqual(base._storage_path, "test/path")
+
 
+    def test_transmit_from_storage_failed(self):
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py:93]: If this is the replacement for failed_not_retryable, shouldn't we call it that to be more specific?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py [hunk 3]
```diff
@@ -81,49 +87,61 @@ def test_constructor(self):
...
-        self.assertEqual(len(os.listdir(self._base.storage._path)), 0)
+        exporter = BaseExporter()
+        exporter.storage = mock.Mock()
+        blob_mock = mock.Mock()
+        blob_mock.lease.return_value = True
+        envelope_mock = {"name":"test","time":"time"}
+        blob_mock.get.return_value = [envelope_mock]
+        exporter.storage.gets.return_value = [blob_mock]
+        with mock.patch("azure.monitor.opentelemetry.exporter.export._base._is_retryable_code"):
+            with mock.patch.object(AzureMonitorClient, 'track', throw(HttpResponseError)):
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py:121]: Can't we combine these 2 with statements?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py (hunk 2 above)
Comment (lzchen) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py:93]: Actually that is the wrong description. Renamed it to the success case.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py (hunk 3 above)
Comment (lzchen) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py:121]: I'm not sure of how the syntax of that looks and don't think it's a big deal.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py (hunk 1 above)
Comment (lzchen) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py:106]: This is the success case in which the stored blob will call `delete` if the transmit succeeded, which we are trying to mock replicate with `TrackResponse` -> https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_generated/operations/_azure_monitor_client_operations.py#L114

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py (hunk 3 above)
Comment (jeremydvoss) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py:120]: Not a big deal, but u can just use a comma. https://stackoverflow.com/questions/893333/multiple-variables-in-a-with-statement

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py (hunk 1 above)
Comment (jeremydvoss) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py:105]: Oh I see. So this is saying, when AzureMonitorClient.track is called, return this. Got it.

**Review by jeremydvoss — APPROVED** [review]

**Pull request merged**